packages/fcl/src/app-utils/verify-signatures.js

```diff
@@ -0,0 +1,186 @@
+import {config} from "@onflow/config"
+import {invariant} from "@onflow/util-invariant"
+import {query} from "../exec/query"
+import {encodeAccountProof} from "../wallet-utils"
+
+const ACCOUNT_PROOF = "ACCOUNT_PROOF"
+const USER_SIGNATURE = "USER_SIGNATURE"
+
+export const validateArgs = args => {
+  if (args.appIdentifier) {
+    const {appIdentifier, address, nonce, signatures} = args
+    invariant(
+      typeof appIdentifier === "string",
+      "appIdentifier must be a string"
+    )
+    invariant(/^[0-9a-f]+$/i.test(nonce), "Nonce must be a hex string")
+    invariant(
+      Array.isArray(signatures) &&
+        signatures.every((sig, i, arr) => sig.f_type === "CompositeSignature"),
+      "Must include an Array of CompositeSignatures to verify"
+    )
+    invariant(
+      signatures.map(cs => cs.addr).every((addr, i, arr) => addr === arr[0]),
+      "User signatures to be verified must be from a single account address"
+    )
+    return true
+  } else {
+    const {message, compSigs} = args
+    invariant(
+      /^[0-9a-f]+$/i.test(message),
+      "Signed message must be a hex string"
+    )
+    invariant(
+      Array.isArray(compSigs) &&
+        compSigs.every((sig, i, arr) => sig.f_type === "CompositeSignature"),
+      "Must include an Array of CompositeSignatures to verify"
+    )
+    invariant(
+      compSigs.map(cs => cs.addr).every((addr, i, arr) => addr === arr[0]),
+      "User signatures to be verified must be from a single account address"
+    )
+    return true
+  }
+}
+
+const getVerifySignaturesScript = async (sig, opts) => {
+  let contractAddress
+  const verifyFunction =
+    sig === "ACCOUNT_PROOF"
+      ? "verifyAccountProofSignatures"
+      : "verifyUserSignatures"
+  const network = await config.first(["env", "network"])
```

Comment:
Should this have a default?

Reply:
Please add a note to the changelog that network is now required to use this api.

Reply:
👍🏼